datafusion-cli/src/print_format.rs

```diff
@@ -244,7 +290,7 @@ mod tests {
             "| 1 |",
             "| 2 |",
             "| 3 |",
-            "+---+",
+            "+---+\n",
```

Comment:
FYI, this is not a real change in the behavior being tested. Before this PR, we used to append the final newline "outside" at the final print out stage. With this PR, since we operate on writers instead of building strings, the final newline is here.